File: Plugins/Orimath.Basics.View/ViewModels/InstructionPointViewModel.cs
using Mvvm;
using Orimath.Plugins;
using Orimath.FoldingInstruction;

namespace Orimath.Basics.View.ViewModels
{
    public class InstructionPointViewModel : NotifyPropertyChanged
    {
        private readonly IViewPointConverter _pointConverter;

        private double _x;
        public double X { get => _x; set => SetValue(ref _x, value); }

        private double _y;
        public double Y { get => _y; set => SetValue(ref _y, value); }

        private InstructionColor _color;
        public InstructionColor Color { get => _color; set => SetValue(ref _color, value); }

        public InstructionPointViewModel(IViewPointConverter pointConverter)
        {
            _pointConverter = pointConverter;
        }

        public InstructionPointViewModel(IViewPointConverter pointConverter, InstructionPoint model)
            : this(pointConverter)
        {
            SetModel(model);
        }

        public void SetModel(InstructionPoint model)
        {
            var point = _pointConverter.ModelToView(model.Point);
            X = point.X;
            Y = point.Y;
            Color = model.Color;
        }
    }
}

File: Plugins/Orimath.Basics.View/ViewModels/InstructionPointViewModel.fs
namespace Orimath.Basics.View.ViewModels
open Orimath.Controls
open Orimath.Plugins
open Orimath.Combination
open ApplicativeProperty

type InstructionPointViewModel(pointConverter: IViewPointConverter) =
    inherit NotifyPropertyChanged()
    member val X = Prop.value 0.0
    member val Y = Prop.value 0.0
    member val Color = Prop.value InstructionColor.Black

    member this.SetModel(model: InstructionPoint) =
        let p = pointConverter.ModelToView(model.Point)
        this.X.Value <- p.X
        this.Y.Value <- p.Y
        this.Color.Value <- model.Color

    static member Create(pointConverter, model) =
        let vm = InstructionPointViewModel(pointConverter)
        vm.SetModel(model)
        vm
